packages/vite/src/node/plugins/define.ts

```diff
@@ -53,16 +53,16 @@ export function definePlugin(config: ResolvedConfig): Plugin {
     }
 
     const replacements: Record<string, string> = {
-      ...processNodeEnv,
+      ...(!ssr ? processNodeEnv : {}),
```

Comment:
Lets only remove the `processNodeEnv` if the `ssr.target` is not `webworker`. Maybe it is needed in that case and it is not needed for the Vitest fix. @frandiox would you also check this one?

Reply:
It also pass in vitest after I add this condition. In the webworker is had other processNodeEnv, so we should replace it?

Reply:
Yes, I think let's keep it until we get a proper report that it isn't working in webworker. Maybe @frandiox knows more about it and we can remove the condition in another PR.

Reply:
ok thanks

Reply:
`process.env` should be replaced in workers because there's no global `process`, so I think this code should work?
Thanks for tagging @patak-dev